components/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,494 @@
+    editPost: (input: Partial<Post>) => Promise<Post>;
+    unsetEditingPost: () => void;
+    openModal: (input: OpenModal) => void;
+    scrollPostListToBottom: () => void;
+}
+
+export type Props = {
+    canEditPost?: boolean;
+    canDeletePost?: boolean;
+    readOnlyChannel?: boolean;
+    channelId: string;
+    codeBlockOnCtrlEnter: boolean;
+    ctrlSend: boolean;
+    config: {
+        EnableEmojiPicker?: string;
+        EnableGifPicker?: string;
+    };
+    maxPostSize: number;
+    useChannelMentions: boolean;
+    editingPost: {
+        post: Post | null;
+        postId?: string;
+        refocusId?: string;
+        title?: string;
+        isRHS?: boolean;
+    };
+    actions: Actions;
+};
+
+export type State = {
+    editText: string;
+    caretPosition: number;
+    postError: React.ReactNode;
+    errorClass: string | null;
+    showEmojiPicker: boolean;
+    renderScrollbar: boolean;
+    scrollbarWidth: number;
+    prevShowState: boolean;
+};
+
+const {KeyCodes} = Constants;
+
+const TOP_OFFSET = 0;
+const RIGHT_OFFSET = 10;
+
+const EditPost = ({editingPost, actions, ...rest}: Props): JSX.Element | null => {
+    const [editText, setEditText] = useState<string>(
+        editingPost.post?.message_source || editingPost.post?.message || '',
+    );
+    const [caretPosition, setCaretPosition] = useState<number>(editText.length);
+    const [postError, setPostError] = useState<React.ReactNode | null>(null);
+    const [errorClass, setErrorClass] = useState<string>('');
+    const [showEmojiPicker, setShowEmojiPicker] = useState<boolean>(false);
+    const [renderScrollbar, setRenderScrollbar] = useState<boolean>(false);
+
+    const textboxRef = useRef<TextboxClass>(null);
+    const emojiButtonRef = useRef<HTMLButtonElement>(null);
+    const wrapperRef = useRef<HTMLDivElement>(null);
+
+    const {formatMessage} = useIntl();
+
+    const shouldScroll = (elementRect: DOMRect, containerRect: DOMRect): boolean => {
+        if (!elementRect || !containerRect) {
+            return false;
+        }
+
+        return elementRect.top < containerRect.top || elementRect.bottom > containerRect.bottom;
+    };
+
+    useEffect(() => {
+        const className = `post-list__dynamic${editingPost.isRHS ? '--RHS' : ''}`;
+        const postListElement = document.getElementsByClassName(className)[0];
+
+        const elementCR = wrapperRef?.current?.getBoundingClientRect();
+        const parentCR = postListElement?.getBoundingClientRect();
+
+        if (elementCR && parentCR) {
+            const shouldScrollIntoView = shouldScroll(elementCR, parentCR);
+
+            if (shouldScrollIntoView) {
+                const scrollOptions: boolean|ScrollIntoViewOptions = isFirefox() ? {behavior: 'smooth', block: 'nearest'} : false;
+                wrapperRef?.current?.scrollIntoView(scrollOptions);
+            }
+        }
+
+        textboxRef?.current?.focus();
+    }, []);
+
+    useEffect(() => {
```

Comment:
absolutely ... i was wondering the same thing, but having it inside the textbox component itself is hardly going to work, since the content needs to live in the upper components. But the idea of adding in a hook to be used when implementing the `CreatePost`/`CreateComment` components is a very good indeed.

I am going to add a comment on that.

Reply:
BTW @hmhealey: is there already a ticket for porting the textbox component to FC?

Reply:
Not on its own, but it's something I'm considering that we might do as part of a larger cleanup of the CreatePost/CreateComment components. We're not planning on that any time soon, but maybe late in the quarter if it doesn't interfere with the effort to make it a rich text editor